src/main.cpp

```diff
@@ -151,188 +237,158 @@ void Frame::OnTagSelectionChange(wxCommandEvent &event)
 
     } catch (const std::exception& e) {
-        std::cerr << e.what() << std::endl;
+        fmt::print("{}\n", e.what());
     }
 
 }
 
 void Frame::OnEntryClick(wxCommandEvent &event)
 {
     const Entry& entry = entry_data.at(event.GetId());
-
-    std::filesystem::path filepath = entry.filepath;
-    if (!std::filesystem::is_directory(filepath)) {
-        // Append "/.." to filepath to access parent directory.
-        filepath = filepath.parent_path();
-    }
-
-    std::string command;
+    const auto filepath = is_directory(entry.filepath) ? entry.filepath : entry.filepath.parent_path();
 
 # if defined (_WIN32)
-    command = "explorer.exe \"" + filepath.string() + "\"";
+    static constexpr auto fmt_str = "explorer.exe \"{}\"";
 # elif defined (__APPLE__)
-    command = "open '" + filepath.string() + "'";
+    static constexpr auto fmt_str = "open '{}'";
 # elif defined (__linux__)
-    command = "xdg-open '" + filepath.string() + "'";
+    static constexpr auto fmt_str = "xdg-open '{}'";
 # else
 #   error "Platform is not supported; can not implement file-opening feature."
 # endif
 
-    std::cout << command << std::endl;
+    const auto command = fmt::format(fmt_str, filepath.string());
+    fmt::print("{}\n", command);
     system(command.data());
 
-    std::cout << entry.filepath << " - " << entry.index << std::endl;
+    fmt::print("{} - {}\n", entry.filepath.string(), entry.index);
 }
 
 Frame::Frame()
-    : wxFrame(NULL, 0, "C-NOTE")
+    : wxFrame(nullptr, 0, "C-NOTE")
 {
-    SetFont(GetFont().Scale(1.5));
+    Frame::SetFont(GetFont().Scale(1.5));
 
     // Loop over all files in the current directory.
-    auto dir = std::filesystem::directory_iterator(".");
-    for (const auto& dir_it : dir) {
+    for (const auto& dir_it : fs::directory_iterator(".")) {
         const auto& path = dir_it.path();
         if (is_directory(path)) {
             // TODO: Loop over all files in all subdirectories.
         } else if (is_regular_file(path)) {
             // Check for tags at beginning.
 
-            //std::cout << path << " is a regular file." << std::endl;
-
-            constexpr size_t contents_max_size = 1024;
-            std::string contents;
-            contents.reserve(contents_max_size);
+            //fmt::print("{} is a regular file.\n", path.string());
+            static constexpr size_t contents_max_size = 1024;
+            static constexpr auto whitespace = "\r\n \t\v"sv;
+            static constexpr auto tag_marker = "#:"sv;
 
-            FILE* file = fopen(path.string().c_str(), "r");
-            size_t bytes_read = fread(contents.data(), 1, contents_max_size, file);
+            File file{path};
+            auto contents = file.read(contents_max_size);
 
-            std::string whitespace("\r\n \t\v");
-            std::string tag_marker("#:");
-
-            char* tags_string = contents.data();
+            auto sv = std::string_view{contents};
```

Comment:
We use a `std::string_view` for parsing because you *can* actually resize a string view.